### DISCUSSION TITLE
InteractiveServerRenderMode prerendering true Datagrid data pre-loading

### DISCUSSION THREAD

**Discussion by ghost** [Megabit/Blazorise#5305] [Q&A]
Hi,

What I try to achieve is to load my data as soon as possible, so the initial rendering of the page is fastest.

I was wondering if there is something I can do better.. I have spend quite some time on it and I am convinced I do something wrong.

* .NET 8
* Blazorise DataGrid 1.4.2
* InteractiveServerRenderMode 

The most successful attempt is with InteractiveServerRenderMode prerendering false. Using the OnInitializedAsync method to start loading data into a variable asap, while the page rendering continues. I am setting the data with Data="@Data" in the control. I don't set the OnRead method, now..  The results is not as fast as with prerendering true, but it doesn't load data twice. 

So, I have to set the InteractiveServerRenderMode prerendering to false, because otherwise the grid data is load twice. Once for the initial rendering, second for the websockets stuff.. 
It empties the grid and reloads the data again when the websockets thread is handled.. very visible in the browser.

InteractiveServerRenderMode prerendering to true had the fastest rendering results, and was perfect. But reloading data in the grid is not ok. Super confusing..  What is the correct way to start loading data before the grid is rendered with prerendering to true? 

Thanks

**Comment by David-Moreira** [comment]
Hello @renebrandnewday 

This is Blazor specific.
As such I invite you to take a look at Microsoft documentation recommended approach to persist the prerendered state:
https://learn.microsoft.com/en-us/aspnet/core/blazor/components/prerendering-and-integration?view=aspnetcore-7.0&pivots=server#persist-prerendered-state

Let us know if this helps
